framework/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -0,0 +1,147 @@
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    costID  = paramInput.findFirst('costID')
+    self.costID  = costID.value
+    valueID = paramInput.findFirst('valueID')
+    self.valueID = valueID.value
+
+  def inputToInternal(self, currentInp):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      In this case, we only want data objects!
+      @ In, currentInp, list, an object that needs to be converted
+      @ Out, currentInp, DataObject.HistorySet, input data
+    """
+    if len(currentInp) > 1:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor {} expects one input DataObject, but received {} inputs!".'
+                                  .format(self.name,len(currentInp)))
+    currentInp = currentInp[0]
+    if currentInp.type not in ['PointSet']:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor "{}" requires a DataObject input! Got "{}".'
+                                 .format(self.name, currentInput.type))
+    return currentInp
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In, inputIn, DataObject, point set that contains the data to be processed
+      @ Out, paretoFrontierDict, dict, dictionary containning the Pareto Frontier information
+    """
+    inData = self.inputToInternal(inputIn)
+    data = inData.asDataset()
+    sortedData = data.sortby(self.costID)
+
+    coordinates = np.zeros(1,dtype=int)
+    for index,elem in enumerate(sortedData[self.costID].values):
+      if (index>1) and (sortedData[self.valueID].values[index]>sortedData[self.valueID].values[coordinates[-1]]):
+        coordinates = np.append(coordinates,index)
+
+    paretoFrontierData = sortedData.isel(RAVEN_sample_ID=coordinates[0]).to_array().values
+    for index,coord in enumerate(coordinates):
+      if index>0:
+        slicedData = sortedData.isel(RAVEN_sample_ID=coord).to_array().values
+        paretoFrontierData = np.vstack((paretoFrontierData,slicedData))
+
+    paretoFrontierDict = {}
+    for index,varID in enumerate(sortedData.data_vars):
+      paretoFrontierDict[varID] = paretoFrontierData[:,index]
+    return paretoFrontierDict
+
+  def collectOutput(self, finishedJob, output):
+    """
+      Function to place all of the computed data into the output object
+      @ In, finishedJob, JobHandler External or Internal instance, A JobHandler object that is in charge of running this post-processor
+      @ In, output, DataObject.DataObject, The object where we want to place our computed results
+      @ Out, None
+    """
+    evaluation = finishedJob.getEvaluation()
+    if isinstance(evaluation, Runners.Error):
+      self.raiseAnError(RuntimeError, "No available output to collect!")
```

Comment:
remove this if statement....it is not possible anymore (I will try to remove all of them from all the other places)

Reply:
removed

Reply:
removed